src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
```

Comment:
Well... That will ran into the same problem... `sun_path` is only 108 bytes long and you will maybe produce an overflow. Have a look at this example from `man 7 unix`:

``` c
struct sockaddr_un name;

// ...

memset(&name, 0, sizeof(struct sockaddr_un));

/* Bind socket to socket name. */

name.sun_family = AF_UNIX;
strncpy(name.sun_path, SOCKET_NAME,
        sizeof(name.sun_path) - 1);

// ...
```

You should also notice, that it's important to ensure the trailing `'\0'`, since `strncpy(dest, src, n)` does not ensure that, if there is no null byte amoge the first `n` bytes of `src`. In this example this is ensured by the `memset()`.